kommunicateui/src/main/java/com/applozic/mobicomkit/uiwidgets/conversation/richmessaging/adapters/KmFormItemAdapter.java

```diff
@@ -59,19 +64,22 @@
     private KmFormStateModel formStateModel;
     private String messageKey;
     private SparseIntArray validationArray;
+    private SparseArray<KmFormPayloadModel.Options> dropdownFieldArray;
     private static final int VALID_DATA = 2;
     private static final int INVALID_DATA = 1;
 
     private static final int VIEW_TYPE_TEXT_FIELD = 1;
     private static final int VIEW_TYPE_SELECTION = 2;
     private static final int VIEW_TYPE_DATETIME = 3;
+    private static final int VIEW_TYPE_DROPDOWN = 4;
 
-    public static final String DEFAULT_DATE_FORMAT = "dd-MM-yyyy";
+    public static final String DEFAULT_DATE_FORMAT = "dd/MM/yyyy";
```

Comment:
can this format be based on the user's device?
default format in US is mm/dd/yyyy

Reply:
Then we will have to provide a format in the payload JSON as other countries will ask for their own format.

Reply:
Then we will have to provide a format in the payload JSON as other countries will ask for their own format.

Reply:
Done.
It will check if dateFormat is provided in the JSON, if not then it will check for US locale and then default locale. The android system is not returning the correct format using the system locale.

Reply:
i guess on web, it detects from the browser and sets it accordingly. a company might have customers across the globe. so setting one format is not possible. see if this can be done dynamically based on device's locale.

Reply:
Tried using dynamic formats but then in android it is not formatting properly.

Reply:
Done. Changed the format to locale based